### PR TITLE
New Spezi Standard Implementation

### PR DESCRIPTION
**Pull request by niallkehoe**
# New Spezi Standard Implementation

## :recycle: Current situation & Problem
Currently, the SpeziScheduler module lacks support for the new Spezi Standard released in Spezi 0.7.0
*This PR implements the improvements mentioned in [https://github.com/https://github.com/StanfordSpezi/Spezi/pull/73](Pull 73) *

## :bulb: Proposed solution
Update the codebase to the newest Spezi 0.7.0

## :gear: Release Notes 
This feature will allow for greater flexibility when using Spezi Questionnaire.

```
/// an example Standard used for the configuration.
actor ExampleStandard: Standard, ObservableObjectProvider, ObservableObject { }
```

### Related PRs
See [https://github.com/https://github.com/StanfordSpezi/Spezi/pull/73](Pull 73) in Spezi for details

### Testing
Passes all tests

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [✅ ] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Tests/UITests/TestApp/ExampleStandard.swift
Comment (PSchmiedmayer) [inline Tests/UITests/TestApp/ExampleStandard.swift:1]: You can remove this type here, as we don't need to explicitly define it in the `Configuration`.

File: Tests/UITests/TestApp/TestAppDelegate.swift [hunk 1]
```diff
@@ -6,16 +6,17 @@
...
 
 
-typealias TestAppScheduler = Scheduler<TestAppStandard, String>
+typealias TestAppScheduler = Scheduler<String>
 
 
 class TestAppDelegate: SpeziAppDelegate {
     override var configuration: Configuration {
-        Configuration(standard: TestAppStandard()) {
+        Configuration(standard: ExampleStandard()) {
```
Comment (PSchmiedmayer) [inline Tests/UITests/TestApp/TestAppDelegate.swift:19]: You should be able to completely go without a `Standard` here as we don't have any requirements on the standard and can just inject a default instance that is used within Spezi.
```suggestion
        Configuration {
```

File: Tests/UITests/TestApp/TestAppDelegate.swift [hunk 2]
```diff
@@ -6,16 +6,17 @@
 // SPDX-License-Identifier: MIT
 //
 
+import Foundation
```
Comment (PSchmiedmayer) [inline Tests/UITests/TestApp/TestAppDelegate.swift:9]: I don't think that the import statement would be needed here:
```suggestion
```

**Comment by vishnuravi** [comment]
Nice work @niallkehoe!

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
Very nice, some small changes, only a few smaller feedback comments that I added in the files. 

It would also be needed to slightly adjust the documentation and remove all mentions of standards in the comments and .md files. E.g. you can remove the last sentence and code example in `SchedulingATask.md`.

You can also remove the `SchedulerTestsStandard` type in the `SchedulerTests.swift` file.

**Comment by niallkehoe** [comment]
@PSchmiedmayer Interesting that this fails the tests on the Github Action but passes all tests on my machine. Any ideas?
        TestAppUITests.testSchedulerBackgroundNotifications()
	TestAppUITests.testSchedulerNotifications()
	TestAppUITests.testSchedulerNotificationsBeforePermissions()

**Review by PSchmiedmayer — APPROVED** [review]

**Comment by PSchmiedmayer** [comment]
@niallkehoe I have taken a look at the testing issues and it seems like this is related to a runtime crash due to some async overloads of the [`UNUserNotificationCenterDelegate `](https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate) protocol.

I could reproduce the crashes in the UI tests using Xcode 15 and using an iOS 17 simulator.

I have moved the code to the completion-based version of the methods in this commit: https://github.com/StanfordSpezi/SpeziScheduler/pull/24/commits/aa271b81bceaf831ab060c8f00b794edb9245f75

**Pull request merged**